src/refspec.c

```diff
@@ -324,7 +324,7 @@ int git_refspec__dwim_one(git_vector *out, git_refspec *spec, git_vector *refs)
 		for (j = 0; formatters[j]; j++) {
 			git_buf_clear(&buf);
 			if (git_buf_printf(&buf, formatters[j], spec->src) < 0)
-				return -1;
+				goto out_err;
```

Comment:
Sounds reasonable and it would help the analysis as we'd be able to #nodef the macro and treat it as an abort when we run into allocation issues.